nameable.rb

```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
```

Comment:
- [ ] Kindly update this exception handling to be more descriptive and point us to the class that has not implemented the `correct_name` method.

  ```
  raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
  ```